### PR TITLE
not_equal #15699

### PR DESCRIPTION
**Pull request by mustafaa7med**
adding not_equal PyTorch Frontend function

not_equal #15699

### CODE REVIEW

**Comment by mustafaa7med** [comment]
Hello @xoiga123,

It's me again, I would really appreciate your review on this new PR, The last one took more than a week to be reviewed and I'm worried this could be the case here, I've made sure this function isn't duplicated before submitting my PR, Thanks in advance for your efforts and looking forward to hearing back from you.

Moustafa

**Comment by hello-fri-end** [comment]
Hi @mustafaa7med ! Thanks for opening this PR. Would you please add a test for the function you implemented? You can take a look at https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html, if you aren't familiar with writing frontend tests. Thanks :)

**Comment by mustafaa7med** [comment]
Hello @hello-fri-end,

Thanks a lot for the fast reply and I wanted to point out that I mentioned xoiga123 before there was an assignee on the PR (**I don't mean any disrespect**), Primarily because he/she was auditing my last PR

**Please note** I'm not an expert on tests, I tried my best to follow the documentation and I would appreciate your review.

Looking forward to hearing back from you.
Moustafa

File: ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py [hunk 1]
```diff
@@ -949,3 +949,33 @@ def test_torch_unique_consecutive(
         dim=axis,
         test_values=False,
     )
+
+@handle_frontend_test(
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_reduction_ops.py:953]: Since it's an instance method, the test should be written in `test_torch/test_tensor.py` and you should use `@handle_frontend_method` decorator for writing the test. Please read the Frontend Instance Method Testing section in the following link: https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html#frontend-instance-method-tests

**Comment by hello-fri-end** [comment]
> Hello @hello-fri-end,
> 
> Thanks a lot for the fast reply and I wanted to point out that I mentioned xoiga123 before there was an assignee on the PR (**I don't mean any disrespect**), Primarily because he/she was auditing my last PR
> 
> **Please note** I'm not an expert on tests, I tried my best to follow the documentation and I would appreciate your review, however, I've checked the logs and I can't seem to see a reason why those 4 checks failed, the logs don't seem to provide relevant information.
> 
> Looking forward to hearing back from you. Moustafa

Hey @mustafaa7med, no worries about the tagging :) I've left some comments on the PR, please make the required changes and re-request a review. Thanks :)

**Review by hello-fri-end — CHANGES_REQUESTED** [review]

**Comment by mustafaa7med** [comment]
Hello @hello-fri-end ,

Thanks for your detailed feedback, I've made the changes requested and would highly appreciate your review.

Thanks in advance,
Moustafa

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -373,6 +373,10 @@ def new_ones(self, size, *, dtype=None, device=None, requires_grad=False):
 
     def floor(self, *, out=None):
         return torch_frontend.floor(self)
+    
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
+    def not_equal(self, other, *, out=out):
```
Comment (hello-fri-end) [inline ivy/functional/frontends/torch/tensor.py:378]:
```suggestion
    def not_equal(self, other, *, out=None):
```

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Looks like `torch.not_equal` has an alias `torch.ne`. Could you please add this? Refer to Alias Functions section in this [link](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html#frontend-instance-method-tests) for adding the alias.

File: ivy/functional/frontends/torch/tensor.py (hunk 2 above)
Comment (mustafaa7med) [reply ivy/functional/frontends/torch/tensor.py]: @hello-fri-end  thank you for your input, I've applied the changes suggested above by changing ``` out=out ``` to ```out=None``` yet I wanted to double check if i should change the test to the following (since it has an alias) 
```
@handle_frontend_test(
    fn_tree="torch.ne",
    aliases=["torch.not_equal"],
    dtype_and_inputs=helpers.dtype_and_values(
        available_dtypes=helpers.get_dtypes("float"),
        num_arrays=2,
        allow_inf=False,
        shared_dtype=True,
    ),
)
def test_torch_not_equal(
    *,
    dtype_and_inputs,
    on_device,
    fn_tree,
    frontend,
    test_flags,
):
    input_dtype, inputs = dtype_and_inputs
    helpers.test_frontend_function(
        input_dtypes=input_dtype,
        frontend=frontend,
        test_flags=test_flags,
        fn_tree=fn_tree,
        on_device=on_device,
        input=inputs[0],
        other=inputs[1],
    )
    ```

Comment (hello-fri-end) [reply ivy/functional/frontends/torch/tensor.py]: Hey @mustafaa7med, you should not use `handle_frontend_test` decorator for instance methods. Currently, `handle_frontend_method` does not have an `aliases` parameter but it's definitely in our TODO list. For now, just adding `ne = not_equal` alias in the `tensor.py` file should suffice. The test that you have written currently fails on the CI with an error saying `on_device` not found. Could you please fix this? (https://github.com/unifyai/ivy/actions/runs/5040147031/jobs/9038722382?pr=15701#step:3:431)

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -8420,12 +8420,55 @@ def test_torch_instance_is_complex(
     input_dtype, x = dtype_and_x
     helpers.test_frontend_method(
         init_input_dtypes=input_dtype,
-        init_all_as_kwargs_np={"data": x[0]},
-        method_input_dtypes=input_dtype,
-        method_all_as_kwargs_np={},
+
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8423]: Why did you delete this?

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 4]
```diff
@@ -8420,12 +8420,55 @@ def test_torch_instance_is_complex(
     input_dtype, x = dtype_and_x
     helpers.test_frontend_method(
         init_input_dtypes=input_dtype,
-        init_all_as_kwargs_np={"data": x[0]},
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8423]: Why did you delete this?

**Comment by hello-fri-end** [comment]
Please take a look at these logs to fix the lining errors : https://github.com/unifyai/ivy/actions/runs/5046109263/jobs/9051200271?pr=15701

**Comment by mustafaa7med** [comment]
Hello @hello-fri-end, 

I've solved the lint formatting issue, Looking forward to hearing back from you.

Moustafa

File: .idea/ivy.iml [hunk 5]
```diff
@@ -2,7 +2,7 @@
 <module type="PYTHON_MODULE" version="4">
   <component name="NewModuleRootManager">
     <content url="file://$MODULE_DIR$" />
-    <orderEntry type="jdk" jdkName="Remote Python 3.8.10 Docker (unifyai/multicuda:base_and_requirements) (2)" jdkType="Python SDK" />
```
Comment (hello-fri-end) [inline .idea/ivy.iml:5]: why is this modified?

File: .idea/vcs.xml [hunk 6]
```diff
@@ -2,6 +2,7 @@
 <project version="4">
   <component name="VcsDirectoryMappings">
     <mapping directory="$PROJECT_DIR$" vcs="Git" />
-    <mapping directory="$PROJECT_DIR$/ivy_tests/test_array_api" vcs="Git" />
```
Comment (hello-fri-end) [inline .idea/vcs.xml:5]: And this?

File: .idea/misc.xml [hunk 7]
```diff
@@ -3,8 +3,8 @@
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
-  <component name="ProjectRootManager" version="2" project-jdk-name="Remote Python 3.8.10 Docker (unifyai/ivy:latest)" project-jdk-type="Python SDK" />
```
[the remark this answers is not in the document]
Comment (mustafaa7med) [reply .idea/misc.xml]: @hello-fri-end, Black formatting made the changes to these files locally yet i reverted the changes.

File: .idea/vcs.xml (hunk 6 above)
Comment (mustafaa7med) [reply .idea/vcs.xml]: @hello-fri-end, Black formatting made the changes to these files locally yet i reverted the changes.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 8]
```diff
@@ -8466,3 +8465,46 @@ def test_torch_instance_copysign(
...
+    init_tree="torch.tensor",
+    method_name="not_equal",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=2,
+        min_value=-1e04,
+        max_value=1e04,
+        allow_inf=False,
+    ),
+    alpha=st.floats(min_value=-1e04, max_value=1e04, allow_infinity=False),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8482]: What is `alpha`? `not_equal` does not have an argument called `alpha`.

File: ivy/functional/frontends/torch/tensor.py [hunk 9]
```diff
@@ -374,6 +374,12 @@ def new_ones(self, size, *, dtype=None, device=None, requires_grad=False):
     def floor(self, *, out=None):
         return torch_frontend.floor(self)
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
```
Comment (hello-fri-end) [inline ivy/functional/frontends/torch/tensor.py:377]:
```suggestion
    @with_unsupported_dtypes({"2.0.1 and below": ("float16","bfloat16")}, "torch")
```
